python/ray/ml/trainer.py

```diff
@@ -139,7 +139,22 @@ def __init__(
         resume_from_checkpoint: Optional[Checkpoint] = None,
     ):
 
-        raise NotImplementedError
+        self.scaling_config = scaling_config if scaling_config else {}
+        self.run_config = run_config if run_config else {}
+        self.datasets = datasets if datasets else {}
+        self.preprocessor = preprocessor
+        self.resume_from_checkpoint = resume_from_checkpoint
+
+    def __new__(cls, *args, **kwargs):
+        """Store the init args as attributes so this can be merged with Tune hparams."""
+        trainer = super(Trainer, cls).__new__(cls)
+        parameters = inspect.signature(cls.__init__).parameters
+        parameters = list(parameters.keys())
+        # Remove self.
+        parameters = parameters[1:]
+        arg_dict = dict(zip(parameters, args))
+        trainer._param_dict = {**arg_dict, **kwargs}
```

Comment:
Would `run_config` also end up being captured here, which should not be merged with Tune hparams.

Reply:
Hmm `run_config` should never be in the Tune hparams in the first place right?